File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditDeleteQuestionAnswerEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.api.question.AnsweredQuestionDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditDeleteQuestionAnswerEventDto extends OrganizationEventDto {
    private AnsweredQuestionDto answeredQuestion;

    OrganizationEditDeleteQuestionAnswerEventDto() {
    }

    public OrganizationEditDeleteQuestionAnswerEventDto(OrganisationId organisationId, AnsweredQuestionDto answeredQuestion) {
        super(organisationId);
        this.answeredQuestion = answeredQuestion;
    }

    public AnsweredQuestionDto getAnsweredQuestion() {
        return answeredQuestion;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDeleteQuestionAnswerEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.question.AnsweredQuestionDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditDeleteQuestionAnswerEventDto(
    override val organizationId: OrganizationId,
    val answeredQuestion: AnsweredQuestionDto,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}